### ISSUE TITLE
MeshBuilder::with_prim_type does nothing

### ISSUE THREAD

**Issue by sadmac7000** [amethyst/rendy#276]
No matter what is passed to with_prim_type, the primitive type is always Triangles.

I'm not that familiar with the code base, but I poked around and from what I can see the primitive type isn't attached to anything. It's assigned in the mesh but never really used.

**Comment by zakarumych** [comment]
Yeah. That's because pipeline dictates primitive type. `Mesh` can't do anything more than checking that pipeline was built with correct primitive type.

**Comment by sadmac7000** [comment]
I'm coming to see that. This is a bug in amethyst_renderer, not rendy specifically. It pushes everything through the same pipeline.
